src/main/kotlin/com/openlattice/mechanic/upgrades/V3StudyMigrationUpgrade.kt

```diff
@@ -0,0 +1,178 @@
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.postgres.mapstores.EntitySetMapstore
+import com.openlattice.postgres.mapstores.PropertyTypeMapstore
+
+import com.hazelcast.query.Predicates
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import com.zaxxer.hikari.HikariDataSource
+
+import java.sql.Connection
+import java.time.OffsetDateTime
+import java.util.EnumSet
+import java.util.Optional
+import java.util.UUID
+
+class V3StudyMigrationUpgrade(
+    toolbox: Toolbox,
+    private val hds: HikariDataSource,
+    private val dataQueryService: PostgresEntityDataQueryService
+): Upgrade {
+
+    private val logger = LoggerFactory.getLogger(V3StudyMigrationUpgrade::class.java)
+
+    private val propertyTypes = HazelcastMap.PROPERTY_TYPES.getMap(toolbox.hazelcast)
+    private val entitySets = HazelcastMap.ENTITY_SETS.getMap(toolbox.hazelcast)
+    private val organizations = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast)
+
+    private val STRING_ID_FQN = FullQualifiedName( "general.stringid" )
+    private val FULL_NAME_FQN = FullQualifiedName( "general.fullname" )
+    private val DESCRIPTION_FQN = FullQualifiedName("diagnosis.Description")
+    private val LAT_FQN = FullQualifiedName("location.latitude")
+    private val LON_FQN = FullQualifiedName("location.longitude")
+    private val SHARING_NAME_FQN =FullQualifiedName("sharing.name")
+    private val VERSION_FQN = FullQualifiedName( "ol.version")
+    private val EMAIL_FQN = FullQualifiedName("contact.Email")
+
+    private val fqnToColumnName = mapOf(
+        STRING_ID_FQN to  "study_id",
+        FULL_NAME_FQN to "fullname",
+        DESCRIPTION_FQN to "description",
+        LAT_FQN to "latitude",
+        LON_FQN to "longitude",
+        SHARING_NAME_FQN to "study_group",
+        VERSION_FQN to "study_version",
+        EMAIL_FQN to "contact",
+        LAST_WRITE_FQN to "updated_at"
+    )
+
+    // private val GET_V2_STUDIES_SQL = """
+    //     SELECT id, ? AS ? FROM
+    //     (SELECT id, ? AS ? FROM data WHERE entity_set_id = ANY(?) AND property_type_id = ? ORDER BY id) AS ?
+    // """.trimIndent()
+
+    // private val V2_DATA_INNERJOIN_SQL = """
+    //     INNER JOIN
+    //     (SELECT id, ? AS ? FROM data WHERE entity_set_id = ANY(?) AND property_type_id = ? ORDER BY id) AS ?
+    //     USING (id)
+    // """.trimIndent()
+
+    // private val INSERT_INTO_STUDY_TABLE_SQL = """
+    //     INSERT INTO studies (id, fullname, description, latitude, longitude, sharing, version, email) VALUES (?,?,?,?,?,?,?,?)
+    // """.trimIndent()
+
+    override fun upgrade(): Boolean {
+        logger.info("starting migration of studies to v3")
+
+        // Property Types of ol.study
+        val studiesPropertyTypes = propertyTypes.entrySet(
+            Predicates.`in`<UUID, PropertyType>(
+                "id",
```

Comment:
`"id"` is not `UUID` so I don't think this will work.

Reply:
`"id"` refers to the `id` of `AbstractSecurableObject` (of which `PropertyType` is an extension of) which is of `UUID` type. It wouldn't matter though since I won't be using `Predicates` as per your comment above.